src/diffusers/models/attention_processor.py

```diff
@@ -2185,9 +2188,19 @@ def __call__(
         hidden_states = torch.bmm(attention_probs, value)
         hidden_states = attn.batch_to_head_dim(hidden_states)
 
+        if masks is not None:
+            if not isinstance(masks, list):
+                masks = [masks]
```

Comment:
```suggestion
            if not isinstance(masks,np.ndarray) or mask.ndim != 4:
                raise ValueError(" ip_adapter_mask should be a numpy array with shape num_ip_adapter, 1, height, width. Please use `IPAdapterMaskProcessor` to preprocess your mask") 
```

let's enforce the `masks` to be a numpy array with shape `[num_ip_adapter, 1,  height, width]`

In order to be able to do so, let's create a `IPAdapterMaskProcessor` that inherits from `VaeImageProcessor` https://github.com/huggingface/diffusers/blob/main/src/diffusers/image_processor.py#L444
and user can get their mask like this

```python
mask_processor = IPAdapterMaskProcessor()
masks = mask_processor.process([mask1,mask2])
```

Reply:
As a reference, this is how I use the `VaeImageProcessor` to process our mask; 
also` IPAdapterMaskProcesso`r should only need to handle mask that has one color, e.g. the masks used in this example, and it does not need to be able to handle [this](https://huggingface.co/datasets/YiYiXu/testing-images/blob/main/ip_mask.png)

```python
import torch
from diffusers import AutoPipelineForText2Image, DDIMScheduler
from transformers import CLIPVisionModelWithProjection
from diffusers.utils import load_image
from diffusers.image_processor import VaeImageProcessor

image_encoder = CLIPVisionModelWithProjection.from_pretrained(
    "h94/IP-Adapter", 
    subfolder="models/image_encoder",
    torch_dtype=torch.float16,
)

pipeline = AutoPipelineForText2Image.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    torch_dtype=torch.float16,
    image_encoder=image_encoder,
)
pipeline.scheduler = DDIMScheduler.from_config(pipeline.scheduler.config)
pipeline.load_ip_adapter(
  "h94/IP-Adapter", 
  subfolder="sdxl_models", 
  weight_name=["ip-adapter-plus-face_sdxl_vit-h.safetensors"] * 2
)
pipeline.set_ip_adapter_scale([0.7] * 2)
pipeline.enable_model_cpu_offload()

face_image1 = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_girl1.png")
face_image2 = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_girl2.png")
mask1 = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_mask1.png")
mask2 = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_mask2.png")

mask_processor = VaeImageProcessor(do_normalize=False, do_binarize=True, do_convert_grayscale=True)
masks = list(mask_processor.preprocess([mask1, mask2]))

generator = torch.Generator(device="cpu").manual_seed(0)

image = pipeline(
    prompt="2 girls",
    ip_adapter_image=[face_image1, face_image2],
    negative_prompt="monochrome, lowres, bad anatomy, worst quality, low quality", 
    num_inference_steps=50, num_images_per_prompt=1,
    generator=generator,
    cross_attention_kwargs={"masks": masks}
).images[0]

```

Reply:
also` IPAdapterMaskProcessor` should only need to handle mask that has one color, e.g. the masks like this

![ip_mask_mask1](https://github.com/huggingface/diffusers/assets/12631849/99f49abe-5f7d-4275-a57a-358371fcd3b3)

And it does not need to be able to handle this kind of mask

![ip_mask](https://github.com/huggingface/diffusers/assets/12631849/8341b5fd-1bd7-4e15-8111-984113a847a0)